### DISCUSSION TITLE
Make AIDriveStrategy hard-coded values editable via XML tags.

### DISCUSSION THREAD

**Discussion by FlexibleGames** [Courseplay/Courseplay_FS22#3346] [Ideas]
I have been trying and trying to make this work but I **hate** LUA and have no idea how to do it elegantly. 

Two values are hard coded in AIDriveStrategy objects (Check out AIDriveStrategyBaler):
                self.slowDownFillLevel = 200
                self.slowDownStartSpeed= 20

These REALLY should be exposed and overloadable via XML. A harvester like the Colossus tries to solve the issue by making the brakes REALLY GOOD, but it doesn't solve it. When the tanks are almost full the faster a vehicle is able to harvest the sooner it **should** start to slow down, but there's no way to edit these values via XML. 200 liters of a buffer is not nearly enough when harvesting at 40 MPH and the harvester AI usually over-runs and misses a sizable block of crops as it tries to stop.

I don't know enough about the DOM of the AIDriveStrategy objects, or how to access various settings (like vehicle/tool harvest speed) or how to augment the getDriveData function that actually uses it. I just copied and tweaked AIDriveStrategyBaler and AIDriveStrategyCombine getDriveData functions and used Utils.overwrittenFunction() to get it to work.

You and the team obviously know LUA and the DOM of the game VERY well and could probably hammer this out in a day. Please consider it.
A couple of options of how to do it, one easy, one far more complex.
The easy way, applies to entire vehicle regardless of FillUnit or configuration values:
```
    <ai>
        <agent slowdownfilllevel="200" slowdownstartspeed="20" width="4.15" length="11.1" height="4.1" lengthOffset="-2" frontOffset="-2.9" />
        <collisionTrigger node="aiCollisionNode" width="4.0" height="4.0"/>
    </ai>
```

The harder way, add multiple options, allowing a configurationset to pick and choose the setting based on the other factors of the vehcile.
```
    <ai>
        <agent width="4.15" length="11.1" height="4.1" lengthOffset="-2" frontOffset="-2.9" />
       <slowdownfilllevels>
            <slowdownfilllevel value="200" slowdownstartspeed="20" />
            <slowdownfilllevel value="2000 slowdownstartspeed="20" />
            <slowdownfilllevel value="8000 slowdownstartspeed="20" />
      </slowdownfilllevels> 
        <collisionTrigger node="aiCollisionNode" width="4.0" height="4.0"/>
    </ai>
```

IMHO this should be part of the base game to give the most tweakable options to the vehicle. Thanks for all your hard work!

**Comment by Tensuko** [comment]
They are hard coded for reasons.
Will not make them editable!
Harvesting at 40km/h is also unrealistic and there won't be support from our side.

**Reply by FlexibleGames** [reply]
Really? Unrealistic? So is planting every kind of crop with the same seed. It's a game, if I wanted to farm for real, I'd buy some land. But okay, I get it, you have your boundaries.
I just wish I knew LUA better so I could do it right.
I prepend the function but it doesn't 'take', like I have to call the original function inside my little code snippet. Is there any documentation anywhere on the Utils functions and how to use them?
